tests/src/Context/MessageContext.php

```diff
@@ -28,7 +29,276 @@ public function iShouldSeeThisFollowingErrorMessage(PyStringNode $message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertErrorVisible($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_error_message_selector');
+      $this->assert(
+        $message,
+        $selector,
+        "The page '%s' does not contain any error messages",
+        "The page '%s' does not contain the error message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertNotErrorVisible($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertNotErrorVisible($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_error_message_selector');
+      $this->assertNot(
+        $message,
+        $selector,
+        "The page '%s' contains the error message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertSuccessMessage($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertSuccessMessage($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_success_message_selector');
+      $this->assert(
+        $message,
+        $selector,
+        "The page '%s' does not contain any success messages",
+        "The page '%s' does not contain the success message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertNotSuccessMessage($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertNotSuccessMessage($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_success_message_selector');
+      $this->assertNot(
+        $message,
+        $selector,
+        "The page '%s' contains the success message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
+  public function assertWarningMessage($message) {
+    if ($this->isCurrentPageAdmin()) {
+      // If it is an admin page, then the admin theme is used and the default
+      // display of messages is applied.
+      parent::assertWarningMessage($message);
+    }
+    else {
+      $selector = $this->getDrupalSelector('front_warning_message_selector');
+      $this->assert(
+        $message,
+        $selector,
+        "The page '%s' does not contain any warning messages",
+        "The page '%s' does not contain the warning message '%s'"
+      );
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @override
+   */
```

Comment:
I will change but do not use a Drupal reference in project that does not concern Drupal directly.